### PR TITLE
Adding helm chart options for psp enabled clusters

### PR DESCRIPTION
**Pull request by stekole**
Signed-off-by: Stefan Kolesnikowicz <stefan.kolesnikowicz@achievers.com>

### What problem does this PR solve?
Kubernetes clusters with PSP (podsecuritypolicy) enabled will not allow chaos-daemon to start up. 

### What is changed and how does it work?

I added a service account, rbac policy and a pod security policy to the chaos-daemon. This allows more access to the pods on clusters which have pod security policies enabled. 

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ X ] Manual test (add detailed scripts or steps below)
- [ ] No code

Tested with and without podsecurity policies enabled.

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by stekole** [comment]
Sorry for the issues submitting this PR. Its my first time. I think I am sorted.

**Comment by STRRL** [comment]
Hi @stekole , thanks for your contribution!  😁

Here is a failed github action that blocks this PR. It is caused by consistent validation between `install.sh` and helm files. 

You could follow below steps to fix it:
- execute `./hack/update_install_script.sh`
- `install.sh` will be updated, execute `make check`, it will do these checks locally
- commit changes

**Comment by stekole** [comment]
@STRRL  - Thanks for the help. I am unsure the failure now. Any recommendations?

**Comment by STRRL** [comment]
Hi @stekole , here is a annotation like `controller-gen.kubebuilder.io/version: v0.2.4`.
It should be `controller-gen.kubebuilder.io/version: v0.2.5`.

You could re-install `controller-gen` versioned v0.2.5 with `go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.5`; then execute `make check` and commit the changes.

**Comment by stekole** [comment]
awesome thanks!

**Review by YangKeao — DISMISSED** [review]
LGTM

**Review by fewdan — APPROVED** [review]
LGTM

File: helm/chaos-mesh/templates/chaos-daemon-rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: chaos-daemon
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+spec:
+  privileged: false
+  allowPrivilegeEscalation: true
+  allowedCapabilities:
+    - '*'
+  runAsUser:
```
Comment (STRRL) [inline helm/chaos-mesh/templates/chaos-daemon-rbac.yaml:77]: seems duplicated with lines 91 - 94

**Review by STRRL — CHANGES_REQUESTED** [review]
Although this podsecuritypolicy is still a little "over-powerd", it represents enough privilledges what we need.

rest LGTM

File: helm/chaos-mesh/templates/chaos-daemon-rbac.yaml (hunk 1 above)
Comment (stekole) [reply helm/chaos-mesh/templates/chaos-daemon-rbac.yaml:77]: Good eye

**Review by STRRL — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**